sys/kern/src/syscalls.rs

```diff
@@ -753,15 +753,16 @@ fn irq_control(
 ) -> Result<NextTask, UserError> {
     let args = tasks[caller].save().as_irq_args();
 
-    let operation = match args.control {
+    let operation = match args.control & 1 {
         0 => crate::arch::disable_irq,
-        1 => crate::arch::enable_irq,
-        _ => {
-            return Err(UserError::Unrecoverable(FaultInfo::SyscallUsage(
-                UsageError::NoIrq,
-            )))
-        }
+        _ => crate::arch::enable_irq,
     };
+    let also_clear_pending = args.control & 2 != 0;
+    if args.control & !3 != 0 {
```

Comment:
similarly, this is just a personal style preference and you're welcome to tell me i'm wrong, but i'd have done
```suggestion
    let also_clear_pending = args.control & 0b10 != 0;
    if args.control & !0b11 != 0 {
```